observability/src/main/java/io/grpc/observability/interceptors/LogHelper.java

```diff
@@ -300,53 +317,85 @@ void logCancel(
         .setEventLogger(eventLogger)
         .setLogLevel(LogLevel.LOG_LEVEL_DEBUG)
         .setRpcId(rpcId);
-    sink.write(logEntryBuilder.build());
+    sink.write(logEntryBuilder.build(), locationTags, customTags);
   }
 
   GrpcLogRecord.Builder createTimestamp() {
     long nanos = timeProvider.currentTimeNanos();
     return GrpcLogRecord.newBuilder().setTimestamp(Timestamps.fromNanos(nanos));
   }
 
+  private static final Set<String> NEVER_INCLUDED_METADATA = new HashSet<>(
+      Collections.singletonList(
+          // grpc-status-details-bin is already logged in a field of the binlog proto
```

Comment:
binlog proto?

 In any case what's the significance of `NEVER_INCLUDED_METADATA` and `ALWAYS_INCLUDED_METADATA`? Why would you exclude "grpc-status-details-bin" and include "grpc-trace-bin"?

Reply:
Added this change as it was in binary logging. Binary logging's explanation for logging metadata can be found [here](https://github.com/grpc/grpc-java/blob/d4fa0ecc07495097453b0a2848765f076b9e714c/services/src/main/proto/grpc/binlog/v1/binarylog.proto#L162). 
As per our discussion offline, in order to keep the behavior same across Java and GO similar I will keep this changes. 
Also, added a TODO to evaluate later and see if the changes makes sense for Observability and decide to remove or keep these changes based on the evaluation/discussion.
Resolving comments on similar changes for now.